src/decoder/mod.rs

```diff
@@ -185,6 +185,28 @@ impl Default for Limits {
     }
 }
 
+/// Metadata for a ScanImage BigTiff file. The first four bytes of metadata
+/// are always SI's magic number which identifies the image as an SI image,
+/// and thus they're left out of the struct. Requesting the ScanImageMetadata
+/// struct from the Decoder via `decoder.scanimage_metadata()` may result in 
+/// a None if the file is not a ScanImage BigTiff.
+#[derive(Debug, Clone)]
+pub struct ScanImageMetadata {
+    /// Bytes 4 to 8 of the static metadata section
+    pub tif_version: u32,
+    /// Non-varying frame data
+    pub nonvar_frame_data: String,
+    /// ROI group data
+    pub roi_group_data: String,
+}
+
+
+impl ScanImageMetadata {
+    pub fn new(tif_version: u32, nonvar_frame_data: String, roi_group_data: String) -> Self {
+       ScanImageMetadata { tif_version, nonvar_frame_data, roi_group_data }
+    }
```

Comment:
There doesn't need to be any public constructor, unless You want to add encoding support as well?